iib/workers/tasks/opm_operations.py

```diff
@@ -26,31 +30,213 @@
 log = logging.getLogger(__name__)
 
 
-def _gen_port_for_grpc() -> Generator[int, None, None]:
-    """
-    Generate port for gRPC service from range set in IIB config.
+class PortFileLock:
+    """A class representing file-lock used during OPM operations."""
 
-    :raises: IIBError when all ports were already taken
-    """
-    conf = get_worker_config()
-    port_start = conf['iib_grpc_start_port']
-    port_end = port_start + conf['iib_grpc_max_port_tries']
+    def __init__(self, purpose: str, port: int):
+        """
+        Initialize the PortFileLock object.
+
+        :param str purpose: Purpose of the lock
+        :param int port: The port number to be locked
+        """
+        self.purpose = purpose
+        self.port = port
+        self.locked = False
+        self.filename = os.path.join(
+            tempfile.gettempdir(),
+            f'iib_{purpose}_{port}.lock',
+        )
+
+    def __repr__(self):
+        """
+        Return string representation of the PortFileLock Object.
 
-    port_stack = list(range(port_start, port_end))
-    random.shuffle(port_stack)
+        :return: String representation of the PortFileLock Object
+        :rtype: str
+        """
+        return f"PortFileLock(port: {self.port}, purpose: {self.purpose}, locked: {self.locked})"
 
-    log.debug('Get random ports from range [%d, %d)', port_start, port_end)
+    def lock_acquire(self):
+        """
+        Create a file representing port lock.
+
+        Before trying to create a a port-lock file, we try to check if the port is free.
+        """
+        log.debug(f"Attempt to lock port {self.port}.")
```

Comment:
Could you, please, use `%s` notation. 
https://docs.python.org/3/howto/logging.html#optimization